logger.go

```diff
@@ -77,3 +82,44 @@ func (l *Logger) Error(v any) {
 func (l *Logger) Errorf(format string, v ...any) {
 	l.lg.Errorf(l.ctx, format, v...)
 }
+
+// AddRequestAttribute adds an attribute (kv) for the parent request log and returns a reference to the original logger for method chaining purposes
+// If the key matches a reserved key, it will be prefixed with "custom_"
+// If the key already exists, its value is overwritten
+func (l *Logger) AddRequestAttribute(key string, value any) *Logger {
+	l.lg.AddRequestAttribute(key, value)
+
+	return l
+}
+
+// WithAttribute returns an AttributerLogger with the provided kv embedded as a child (trace) log attribute
+// If the key matches a reserved key, it will be prefixed with "custom_"
+// If the key already exists, its value is overwritten
+func (l *Logger) WithAttribute(key string, value any) *AttributerLogger {
```

Comment:
I am not a big fan of `WithAttribute()` taking in a key/value pair. This is a bit inconsistent interface where the first key/value is added by `WithAttribute()` and all the rest are added by `AddAttribute()`.